src/Microsoft.ML.Experimental/DataLoadSave/Database/DbExtensions.cs

```diff
@@ -66,5 +66,82 @@ public static Type ToType(this DbType dbType)
                     return null;
             }
         }
+
+        /// <summary>Maps a <see cref="InternalDataKind"/> to the associated <see cref="DbType"/>.</summary>
+        public static DbType ToDbType(this InternalDataKind dataKind)
+        {
+            switch (dataKind)
+            {
+                case InternalDataKind.I1:
+                {
+                    return DbType.SByte;
+                }
+
+                case InternalDataKind.U1:
+                {
+                    return DbType.Byte;
+                }
+
+                case InternalDataKind.I2:
+                {
+                    return DbType.Int16;
+                }
+
+                case InternalDataKind.U2:
+                {
+                    return DbType.UInt16;
+                }
+
+                case InternalDataKind.I4:
+                {
+                    return DbType.Int32;
+                }
+
+                case InternalDataKind.U4:
+                {
+                    return DbType.UInt32;
+                }
+
+                case InternalDataKind.I8:
+                {
+                    return DbType.Int64;
+                }
+
+                case InternalDataKind.U8:
+                {
+                    return DbType.UInt64;
+                }
+
+                case InternalDataKind.R4:
+                {
+                    return DbType.Single;
+                }
+
+                case InternalDataKind.R8:
+                {
+                    return DbType.Double;
+                }
+
+                case InternalDataKind.TX:
+                {
+                    return DbType.String;
+                }
+
+                case InternalDataKind.BL:
+                {
+                    return DbType.Boolean;
+                }
+
+                case InternalDataKind.DT:
```

Comment:
What about DateTimeOffset (InternalDataKind.DZ) ?

Reply:
There is no equivalent for DbKind

Reply:
Even `DbType.DateTimeOffset` and `DbType.Time`?

https://docs.microsoft.com/en-us/dotnet/api/system.data.dbtype

Reply:
The latter is the SQL time type, not an equivalent to the `System.TimeSpan` type; I'm not sure what the appropriate conversion is. I missed `DateTimeOffset`.

Reply:
> The latter is the SQL time type, not an equivalent to the System.TimeSpan type;

According to https://docs.microsoft.com/en-us/dotnet/framework/data/adonet/sql-server-data-type-mappings, it is the appropriate mapping:


SQL Server Database Engine type | .NET Framework type | SqlDbType enumeration | SqlDataReader SqlTypes typed accessor | DbType enumeration | SqlDataReader DbType typed accessor
-- | -- | -- | -- | -- | --
time(SQL Server 2008 and later) | TimeSpan | Time | none | Time | GetDateTime

Reply:
Is that also the appropriate mapping for all database types, not just SQL?

Also, it looks like it isn't even the SQL `time` type, it is the `SQL` `datetime` type and there is a separate value in a separate enum that should be used for `SQL` `time`:

> | Time | 17 | A type representing a SQL Server DateTime value. If you want to use a SQL Server time value, use [Time](https://docs.microsoft.com/en-us/dotnet/api/system.data.sqldbtype?view=netframework-4.8#System_Data_SqlDbType_Time). |

Reply:
ah, right.
These weren't done because `DbDataReader` doesn't expose `Get` methods for them